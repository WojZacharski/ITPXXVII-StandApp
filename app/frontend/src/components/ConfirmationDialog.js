import React from 'react';
import Dialog from "@material-ui/core/Dialog";
import DialogTitle from "@material-ui/core/DialogTitle";
import DialogContent from "@material-ui/core/DialogContent";
import DialogContentText from "@material-ui/core/DialogContentText";
import DialogActions from "@material-ui/core/DialogActions";
import {useLanguageContext} from "../contexts/LanguageContext";
import {PrimaryButton} from "./buttons/PrimaryButton";
import {SecondaryButton} from "./buttons/SecondaryButton";

export const ConfirmationDialog = ({handleClose, open, handleSave}) => {
    const handleSaveAction = () => {
        handleSave();
        handleClose();
    };
    const {t} = useLanguageContext();

    return <Dialog open={open}
                   onClose={handleClose}
                   aria-labelledby="alert-dialog-title"
                   aria-describedby="alert-dialog-description">
        <DialogTitle id="alert-dialog-title">{t("ARE_YOU_SURE")}</DialogTitle>
        <DialogContent>
            <DialogContentText id="alert-dialog-description">
                {t("AFTER_SUBMIT_INFO")}
            </DialogContentText>
        </DialogContent>
        <DialogActions>
            <SecondaryButton onClick={handleClose} variant="contained">
                {t('DISAGREE')}
            </SecondaryButton>
            <PrimaryButton className='submit' onClick={handleSaveAction} variant="contained" autoFocus>
                {t("SUBMIT")}
            </PrimaryButton>
        </DialogActions>
    </Dialog>;
};