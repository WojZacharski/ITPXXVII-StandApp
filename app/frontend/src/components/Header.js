import React from 'react';
import {useDataContext} from "../contexts/DataContext";
import {useLanguageContext} from "../contexts/LanguageContext";
import {makeStyles} from "@material-ui/core";
import {standTypeResolver} from "../utils/standTypeResolver";

const useStyles = makeStyles(theme => ({
    underline: {
        borderBottom: '2px solid black',
        width: 'auto',
        display: 'inline-block',
        paddingRight: '20px'
    }
}));

export const Header = () => {
    const {tokenData: {companyName = "Unknown", type = "Unknown"} = {}} = useDataContext();
    const {t} = useLanguageContext();
    const classes = useStyles();

    return <div>
        <h1 className={classes.underline}>{t("STAND_SELECTION")}: {companyName} / {standTypeResolver(type, t)}</h1>
    </div>;
};


