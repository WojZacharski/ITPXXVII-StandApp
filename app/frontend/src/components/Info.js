import React from 'react';
import InfoIcon from '@material-ui/icons/Info';
import {useLanguageContext} from "../contexts/LanguageContext";
import {makeStyles} from "@material-ui/core";

const useStyles = makeStyles(theme => ({
    inline: {
        display: 'flex',
        alignItems: 'center'
    },
    infoIcon: {
        color: '#1565c0',
        fontSize: '32px',
        marginRight: '6px'
    }
}));

export const Info = ({saved}) => {
    const {t} = useLanguageContext();
    const classes = useStyles();
    const info = saved
        ? t("CHANGE_INFO")
        : t('INFO');

    return <div className={classes.inline}>
        <InfoIcon className={classes.infoIcon}/>
        <p>{info}</p>
    </div>;
};
