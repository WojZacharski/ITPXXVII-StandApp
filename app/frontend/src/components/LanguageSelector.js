import React from 'react';
import {translations} from "../constants/translations";
import {makeStyles} from "@material-ui/core";
import {useLanguageContext} from "../contexts/LanguageContext";

const useStyles = makeStyles(theme => ({
    option: {
        margin: `${theme.spacing(0.5)}px ${theme.spacing(2)}px`,
        '&:hover': {
            textDecoration: 'underline',
            cursor: 'pointer'
        }
    }
}));

export const LanguageSelector = (props) => {
    const languages = ["Polski", "English"];
    const {setLanguage} = useLanguageContext();
    const classes = useStyles();

    return <div className={props.className}>
        {Object.keys(translations).map((lang, index) =>
            <span className={classes.option} key={index} onClick={() => setLanguage(lang)}>{languages[index]}</span>)}
    </div>
};
