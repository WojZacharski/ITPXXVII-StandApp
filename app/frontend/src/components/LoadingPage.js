import React from 'react';
import InfoIcon from '@material-ui/icons/Info';
import {makeStyles} from "@material-ui/core";

const useStyles = makeStyles(theme => ({
    container: {
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        marginTop: '30vh'
    },
    infoIcon: {
        fontSize: '64px',
        marginRight: '12px',
        color: '#1565c0'
    },
}));

export const LoadingPage = () => {
    const classes = useStyles();

    return <div className={classes.container}>
        <InfoIcon className={classes.infoIcon}/>
        <h1>Page is loading...</h1>
    </div>;
};