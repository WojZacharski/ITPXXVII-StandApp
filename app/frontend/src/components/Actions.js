import React from 'react';
import {useDataContext} from "../contexts/DataContext";
import {ConfirmationDialog} from "./ConfirmationDialog";
import {useLanguageContext} from "../contexts/LanguageContext";
import {Info} from "./Info";
import {PrimaryButton} from "./buttons/PrimaryButton";

export const Actions = () => {
    const {t} = useLanguageContext();
    const {id, selectedStand, saveChoice} = useDataContext();
    const [open, setOpen] = React.useState(false);
    const submitDisabled = !!(selectedStand || !id);

    const handleOpen = () => setOpen(true);
    const handleClose = () => setOpen(false);

    return <div>
        <Info saved={selectedStand}/>
        <PrimaryButton variant="contained" onClick={handleOpen} disabled={submitDisabled}>{t("SUBMIT")}</PrimaryButton>
        <ConfirmationDialog handleClose={handleClose} handleSave={saveChoice} open={open}/>
    </div>
};