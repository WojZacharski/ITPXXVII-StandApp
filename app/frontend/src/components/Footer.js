import React from 'react';
import {makeStyles} from "@material-ui/core";
import {LanguageSelector} from "./LanguageSelector";
import {useLanguageContext} from "../contexts/LanguageContext";

const useStyles = makeStyles(theme => ({
    footer: {
        position: "fixed",
        bottom: 0,
        width: "100%",
        height: "40px",
        display: 'flex',
        alignItems: 'center',
        backgroundColor: '#292c2f',
        color: '#fff',
        flex: 'auto',
        '& > *': {
            flex: 1,
            margin: '16px'
        }
    },
    languageSelector: {
        textAlign: 'center'
    },
    company: {
        color: '#8f9296',
        display: 'block',
    },
    contact: {
        color: '#8f9296',
        textAlign: 'right',
        '& a': {
            textDecoration: 'none',
            color: '#8f9296',
            '&:hover': {
                textDecoration: 'underline'
            }
        }
    }
}));

export const Footer = (props) => {
    const classes = useStyles();
    const {t} = useLanguageContext();

    return <footer className={classes.footer}>
        <p className={classes.company}>Stowarzyszenie Studentów BEST AGH Kraków © 2024</p>
        <LanguageSelector className={classes.languageSelector}/>
        <p className={classes.contact}>{t("CONTACT")}: <a href={'mailto:itp@best.krakow.pl'}>itp@best.krakow.pl</a></p>
    </footer>
};