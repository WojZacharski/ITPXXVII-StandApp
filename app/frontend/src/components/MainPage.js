import React from 'react';
import {Header} from "./Header";
import {Actions} from "./Actions";
import {StandMap} from "./StandMap";
import {makeStyles} from "@material-ui/core";
import Container from "@material-ui/core/Container";

const useStyles = makeStyles(theme => ({
    container: {
        '@media (min-width: 1600px)': {
            zoom: '120%'
        },
        margin: 'auto',
    }
}));

export const MainPage = () => {
    const classes = useStyles();

    return <Container className={classes.container}>
            <Header/>
            <StandMap/>
            <Actions/>
        </Container>
};
