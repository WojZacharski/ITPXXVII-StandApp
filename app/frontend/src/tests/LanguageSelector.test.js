import React from 'react';
import {mount} from 'enzyme';
import {languageData, mockData, MockDataProvider} from "./__mocks__/MockDataProvider";
import {LanguageSelector} from "../components/LanguageSelector";

describe("LanguageSelector Component", () => {
    const component = mount(<MockDataProvider>
        <LanguageSelector/>
    </MockDataProvider>);

    it('should render and match snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    it('should click PL', () => {
        component.find('span').first().simulate('click');
        expect(languageData.setLanguage).toHaveBeenCalledTimes(1);
        expect(languageData.setLanguage).toHaveBeenCalledWith("PL");
    });

    it('should click EN', () => {
        component.find('span').at(1).simulate('click');
        expect(languageData.setLanguage).toHaveBeenCalledTimes(2);
        expect(languageData.setLanguage).toHaveBeenCalledWith("EN");
    });
});