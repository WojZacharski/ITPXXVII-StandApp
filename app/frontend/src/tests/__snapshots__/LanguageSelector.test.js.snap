// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelector Component should render and match snapshot 1`] = `ReactWrapper {}`;
